src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -368,37 +371,36 @@ void Renderer::Impl::render(const RenderTree& renderTree,
     // Drawables
     const auto drawableOpaquePass = [&] {
         const auto debugGroup(parameters.renderPass->createDebugGroup("drawables-opaque"));
-        const auto maxLayerIndex = orchestrator.maxLayerIndex();
         parameters.pass = RenderPass::Opaque;
-        parameters.currentLayer = 0;
-        parameters.depthRangeSize = 1 -
-                                    (maxLayerIndex + 3) * PaintParameters::numSublayers * PaintParameters::depthEpsilon;
+        parameters.depthRangeSize = 1 - (orchestrator.numLayerGroups() + 2) * PaintParameters::numSublayers *
+                                            PaintParameters::depthEpsilon;
 
         // draw layer groups, opaque pass
+        int32_t i = static_cast<int32_t>(orchestrator.numLayerGroups()) - 1;
         orchestrator.visitLayerGroups([&](LayerGroupBase& layerGroup) {
-            parameters.currentLayer = layerGroup.getLayerIndex();
+            parameters.currentLayer = i--;
```

Comment:
It looks like the counting direction is swapped between 3D and opaque passes.

Reply:
Yes, the opaque pass is in reversed order in legacy, but on drawable is not so I had to adjust the index. I will explore drawing opaque layers in reverse order as on legacy in different PR. I think they did it like this to optimize the rendering based on depth test.

Reply:
Interesting that it never caused any problems.

Reply:
It affects the `opaqueCutOff` logic, so the fill-extrusion depth logic. Until now I saw the `currentLayer` it is used only there.